### PR TITLE
feat(mock-tests): Mock Arc Client for testing purposes

### PR DESCRIPTION
**Pull request by kuba-4chain**

### CODE REVIEW

File: broadcast/broadcast-client/client_builder_mocks.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	broadcast_api "github.com/bitcoin-sv/go-broadcast-client/broadcast"
+	"github.com/bitcoin-sv/go-broadcast-client/broadcast/internal/arc/mocks"
+)
+
+type MockType int
+
+const (
+	MockSuccess MockType = iota + 1
```
Comment (arkadiuszos4chain) [inline broadcast/broadcast-client/client_builder_mocks.go:11]: Why `+1`?

Comment (kuba-4chain) [reply broadcast/broadcast-client/client_builder_mocks.go:11]: I don't know, probably could or even should start from 0. For some reason I thought that enums start with 1

File: broadcast/broadcast-client/client_builder_mocks.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package broadcast_client
```
Comment (dorzepowski) [inline broadcast/broadcast-client/client_builder_mocks.go:1]: I would rather use a separated package like `mock_broadcast_client` or `broadcast_client_test` or something that is showing that it's for testing purposes.

And also IMHO we should provide a separated builder for tests.

File: broadcast/broadcast-client/client_builder_mocks.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+const (
+	MockSuccess MockType = iota + 1
+	MockFailure
+	MockTimeout
+	MockAlreadyOnChain
+	MockInMempool
+)
+
+func (cb *builder) WithMockArc(config ArcClientConfig, mockType MockType) *builder {
```
Comment (dorzepowski) [inline broadcast/broadcast-client/client_builder_mocks.go:18]: Thinking out loud:
As we're providing this as a published component, maybe we should also have tests for it, to ensure that it is working as expected. Beside that those tests can be good intro for anyone who would like to use that in his tests.

File: broadcast/broadcast-client/client_builder_mocks.go (hunk 2 above)
Comment (dorzepowski) [reply broadcast/broadcast-client/client_builder_mocks.go:1]: Also it would be useful to add some instructions in the go docs of this new builder.

Comment (kuba-4chain) [reply broadcast/broadcast-client/client_builder_mocks.go]: Done

File: broadcast/broadcast-client/client_builder_mocks.go (hunk 3 above)
Comment (kuba-4chain) [reply broadcast/broadcast-client/client_builder_mocks.go]: Sure, good idea, done

**Review by dorzepowski — APPROVED** [review]

**Pull request merged**